builders/build/substrate-api/py-substrate-interface.md

```diff
@@ -0,0 +1,311 @@
+---
+title: How to use the Python Substrate Interface
+description: Follow this tutorial to learn the basic of how to use the Python Substrate Interface library to query chain data, send transactions, and more on Moonbeam networks.
+---
+
+# Python Substrate Interface
+
+![Intro diagram](/images/builders/build/substrate-api/py-substrate-interface/py-substrate-interface-banner.png)
+
+## Introduction {: #introduction }
+
+[Python Substrate Interface](https://github.com/polkascan/py-substrate-interface){target=_blank} library allows application developers to query a Moonbeam node and interact with the node's Polkadot or Substrate features using a native Python interface. Here you will find an overview of the available functionalities and some commonly used code examples to get you started on interacting with Moonbeam networks using Python Substrate Interface. 
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+Installing and using Python Substrate Interface library requires `pip` to be installed. 
+
+Use the following command to install `pip` for Python 3:
+
+1. Start by updating the package list using the following command:
+```
+sudo apt update
+```
+
+2. Use the following command to install pip for Python 3:
+```
+sudo apt install python3-pip
+```
```

Comment:
these are only linux commands and on macos you don't need to install pip as if you install python3 with homebrew it comes with it, so I don't know if we need this section in steps

also later on in the guide in the sending transactions part people are going to need an account with funds and if they're on moonbeam/moonriver they'll need an endpoint... similar to the Ethers page, so you could just copy the prereqs from there and add pip to the list